### PR TITLE
net: socket: Start service thread from initialization function

### PR DESCRIPTION
**Pull request by jukkar**
We cannot always start the service monitor thread statically because the static threads are started after the application level. This means that the when config library wants to start dhcpv4 server which uses socket services, there would be a deadlock. Simplest solution is to start the service thread directly from socket service init function.

### CODE REVIEW

File: subsys/net/lib/sockets/sockets_service.c [hunk 1]
```diff
@@ -277,15 +277,23 @@ static void socket_service_thread(void)
...
-K_THREAD_DEFINE(socket_service_monitor, CONFIG_NET_SOCKETS_SERVICE_STACK_SIZE,
-		socket_service_thread, NULL, NULL, NULL,
-		CLAMP(CONFIG_NET_SOCKETS_SERVICE_THREAD_PRIO,
-		      K_HIGHEST_APPLICATION_THREAD_PRIO,
-		      K_LOWEST_APPLICATION_THREAD_PRIO), 0, 0);
-
 static int init_socket_service(void)
 {
-	k_thread_name_set(socket_service_monitor, "net_socket_service");
+	static k_tid_t ssm;
```
Comment (rlubos) [inline subsys/net/lib/sockets/sockets_service.c:282]: Should `ssm`this be `static`? We only use thread ID to set the name, I don't think there's a need to keep that memory after.

**Review by pdgendt — DISMISSED** [review]
Nit: commit message has a "when" too much.

File: subsys/net/lib/sockets/sockets_service.c (hunk 1 above)
Comment (jukkar) [reply subsys/net/lib/sockets/sockets_service.c:282]: Right, not need for static. Will fix.

**Comment by jukkar** [comment]
> "when" too much.

Indeed, I will try to formulate better text.

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**